dashboard/app/controllers/foorm_editor_controller.rb

```diff
@@ -1,8 +1,11 @@
+# Foorm Editor is only available on levelbuilder or test, for those with levelbuilder permissions.
 class FoormEditorController < ApplicationController
+  before_action :require_levelbuilder_mode_or_test_env
+  before_action :authenticate_user!
+
   # GET '/foorm/editor/'
   def index
-    # only show for admins on non-production
-    return render_404 if Rails.env.production? || !current_user.admin?
+    return render_404 unless current_user && current_user.levelbuilder?
```

Comment:
this would better be done with CanCan. Here's how we do it for curriculum routes: 
* https://github.com/code-dot-org/code-dot-org/blob/a30a13d14025ec95f71cea7d4d427273b33eb78b/dashboard/app/models/ability.rb#L224
* and then `authorize_resource`

Reply:
done!